README.md

```diff
@@ -52,7 +52,10 @@ giving us both centralized and local logs.  (Redundancy is nice).
 		flag.Parse()
 
 		if graylogAddr != "" {
-			gelfWriter, err := gelf.NewWriter(graylogAddr)
```

Comment:
Oh, I do like those functional options. That's basically a monoid with the operation being function application and the identity element being the identity function (that returns its input). Well, appropriately distorted to take into account that mutable variables are being passed around. That kind of thing is a trick we use in Haskell.

That said, while I can see how this could be a handy trick for wrapping the TCP connection in TLS, since they're both stream connections:
1. I think we'd may still have the library supply a constructor for the TLS connectivity since applications other than `SystemdJournal2Gelf` may want to use TLS; and
2. I can't really see using a common constructor working for stream (TCP/TLS) and datagram (UDP) connections because the options are just so completely different.

The only thing that the stream and datagram connections share in common are the host and port. But datagram has chunking (implying a size limit users may want to set) and (optional) compression, neither of which stream connections allow. Stream connections potentially have all the TLS key and cert info, which datagram connections do not have.